src/waypoint_cmd.cpp

```diff
@@ -161,7 +162,7 @@ extern CommandCost CanExpandRailStation(const BaseStation *st, TileArea &new_ta,
 CommandCost CmdBuildRailWaypoint(TileIndex start_tile, DoCommandFlag flags, uint32 p1, uint32 p2, const char *text)
 {
 	/* Unpack parameters */
-	Axis axis      = Extract<Axis, 4, 1>(p1);
+	Axis axis      = Extract<Axis, 6, 1>(p1);
```

Comment:
It fixes building rail waypoints. By the way, railtype is passed as a param to this command, but it is not used.